src/android/shakeDetector.java

```diff
@@ -157,37 +145,52 @@ else if(event.sensor.getType() == Sensor.TYPE_LINEAR_ACCELERATION){
+                    notifyEvent(this, POTENTIAL_INCIDENT, new JSONObject());
                 }
-                */
+
                 acceleration2 = acceleration2 * 0.85f + (currAcceleration - lastAcceleration);
                 String LAccTag2 = "2Shake0.6LA0.85";
 
-                Log.i(this, LAccTag2, ","+String.valueOf(currTime)+"," + Float.toString(xAcceleration) + "," + Float.toString(yAcceleration) + "," + Float.toString(zAcceleration) + "," + Float.toString(acceleration2) + ",");
-
+                //Log.i(this, LAccTag2, ","+String.valueOf(currTime)+"," + Float.toString(xAcceleration) + "," + Float.toString(yAcceleration) + "," + Float.toString(zAcceleration) + "," + Float.toString(acceleration2) + ",");
 
-                /*
                 if ((acceleration > 13f) && (Math.abs(xAcceleration) < 10)) {
-                    //lastUpdated = currTime;
-                    NotificationHelper.createNotification(this, Constants.TRACKING_ERROR_ID, AccTag2);
-                    notifyEvent(this, LAccTag2, new JSONObject());
-                    //Log.i(this, AccTag2, "," + Float.toString(xAcceleration) + "," + Float.toString(yAcceleration) + "," + Float.toString(zAcceleration) + "," + Float.toString(acceleration2) + ",");
+                    lastUpdated = currTime;
+                    //NotificationHelper.createNotification(this, Constants.TRACKING_ERROR_ID, LAccTag2);
+                    Log.i(this, LAccTag2, "," + Float.toString(xAcceleration) + "," + Float.toString(yAcceleration) + "," + Float.toString(zAcceleration) + "," + Float.toString(acceleration2) + ",");
+                    notifyEvent(this, POTENTIAL_INCIDENT, new JSONObject());
+                }
+
+            }
+            else if(event.sensor.getType() == Sensor.TYPE_GYROSCOPE){
+                float xRotation = event.values[0];
+                float yRotation = event.values[1];
+                float zRotation = event.values[2];
+                String GyroTag = "2Gyro5";
+
+                //Log.i(this, GyroTag, ","+String.valueOf(currTime)+"," + Float.toString(xRotation) + "," );
+
+                if ((xRotation > 5f) && (yRotation < 5f) && (zRotation) < 5f) {
+                    lastUpdated = currTime;
+                    //NotificationHelper.createNotification(this, Constants.TRACKING_ERROR_ID, GyroTag);
+                    Log.i(this, GyroTag, "," + Float.toString(xRotation) + "," );
+                    notifyEvent(this, POTENTIAL_INCIDENT, new JSONObject());
                 }
-                */
             }
         }
     }
 
     public void notifyEvent(Context context, String eventName, JSONObject autogenData) {
-        String TAG = "shakeDetector";
+        String TAG = "ShakeDetector";
+        Intent shakeDetectionIntent = new Intent();
+        shakeDetectionIntent.setAction(eventName);
+        shakeDetectionIntent.putExtras(jsonToBundle(autogenData));
+        context.sendBroadcast(shakeDetectionIntent);
```

Comment:
the newly added code is not really notifying an event, it is broadcasting an intent.
you should really refactor into `broadcastAndNotifyEvent` as I think I did.